docs/developer-setup.md

```diff
@@ -0,0 +1,234 @@
+```
+docker compose exec nextcloud bash --login -c "cd /var/www/html/apps-extra/spreed && make dev-setup"
+```
+
+## Start Nextcloud
+
+In the directory of `nextcloud-docker-dev` execute the following:
+
+```
+docker compose up -d nextcloud
+```
+
+Enable Nextcloud Talk:
+
+```
+./scripts/occ.sh nextcloud -- app:enable spreed
+```
+
+After that you should be able to login with `admin` / `admin` at https://nextcloud.local
+
+### Enable High Performance Backend
+
+If you want to enable the HPB for your development setup, just execute
+
+```
+./scripts/enable-talk-hpb.sh nextcloud
+```
+
+to start the needed containers and add the configuration to the Nextcloud installation. You can check if everything is working in Settings -> Talk -> High Performance Backend.
+
+### Enable Nextcoud Talk Recording Server
+
+**Important:** Make sure that the HPB is correctly running!
+
+Start the container of the recording server:
+
+```
+docker compose up -d talk-recording
+```
+
+Go to the admin settings of talk and add the recording server (`http://talk-recording.local` with shared secret `6789`)
+
+### Use federation between 2 Nextcloud Servers
+
+The provided `docker-compose.yml` file from `nextcloud-docker-dev` supports spinning up multiple instances out of the box. For the `master/main` branch, up to 3 instances can be started (e.g. `docker compose up -d nextcloud nextcloud2 nextcloud3`). This allows easy testing/development for federated scenarios.
+
+1. Make sure to have 2 instances of Nextcloud running with
+
+    ```
+    docker compose up -d nextcloud nextcloud2
+    ```
+
+2. In case of federated calling, the HPB needs to be enabled on both instances
+
+    ```
+    ./scripts/enable-talk-hpb.sh nextcloud
+    ./scripts/enable-talk-hpb.sh nextcloud2
+    ```
+
+3. Allow self signed certs on both instances
+
+    ```
+    ./scripts/occ.sh nextcloud -- config:system:set sharing.federation.allowSelfSignedCertificates --value true --type bool
+    ./scripts/occ.sh nextcloud2 -- config:system:set sharing.federation.allowSelfSignedCertificates --value true --type bool
+    ```
+
+4. Copy and import the certificates to the other instance
+
+    ```
+    docker compose cp data/ssl/nextcloud2.local.crt nextcloud:/tmp
+    docker compose cp data/ssl/nextcloud.local.crt nextcloud2:/tmp
+
+    ./scripts/occ.sh nextcloud2 -- security:certificates:import /tmp/nextcloud.local.crt
+    ./scripts/occ.sh nextcloud -- security:certificates:import /tmp/nextcloud2.local.crt
+    ```
+
+5. Optional: Verify certs
+
+    ```
+    ./scripts/occ.sh nextcloud2 -- security:certificates
+    ./scripts/occ.sh nextcloud -- security:certificates
+    ```
+
+6. Enable federation in the admin settings of Nextcloud Talk.
+
+### Rebuild / update Talk after code changes
+
+#### JavaScript
+
+If you modify any JavaScript file, either directly or by switching to a different branch, you will need to rebuild them. You can do that with:
```

Comment:
Maybe also mention that it's possible to use method 2 with npm package

Reply:
Kids... :shrug: Anyway, done.